task_processing/plugins/kubernetes/task_config.py

```diff
@@ -0,0 +1,43 @@
+import re
+
+from pyrsistent import field
+from pyrsistent import PMap
+from pyrsistent import PVector
+
+from task_processing.interfaces.task_executor import DefaultTaskConfigInterface
+
+
+class KubernetesTaskConfig(DefaultTaskConfigInterface):
+    node_selector = field(type=PMap)
+    containers = field(type=PVector)
+    # Hardcoded for the time being
+    restart_policy = "Never"
+    # By default, the retrying executor retries 3 times. This task option
+    # overrides the executor setting.
+    retries = field(type=int,
+                    factory=int,
+                    mandatory=False,
+                    invariant=lambda r: (r >= 0, 'retries >= 0'))
+    volumes = field(type=PVector)
+
+    @property
+    def pod_name(self) -> str:
+        return self.name
+
+    def set_pod_name(self, pod_name: str):
```

Comment:
looking at this now, i don't think we use set_task_id anywhere, in mesos we create a task with a name and uuid is generated randomly. task_id value is then derived from those two. when state is recovered - it's `name` and `uuid` that are deserialized, not `task_id`.

you can probably just remove this method until we actually find a reason to have it.

Reply:
we use this in tron (https://github.com/Yelp/Tron/blob/5fd0b08ef8c7fd965960b105622c04499406aeb8/tron/mesos.py#L481) - iiuc, this is for recreating tron state after a restart

Reply:
you're right, though if name and uuid are serialized in recovery data, we might still not need it, but it's not a blocker of course, i'm ok with keeping it